<?xml version="1.0"?>

<launch>
   <arg name="model" default="$(find hector_quadrotor_description)/urdf/quadrotor.gazebo.xacro" />

   <group ns="uav1">
     <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
       <arg name="name" value="uav1" />
       <arg name="tf_prefix" value="uav1" />
       <arg name="model" value="$(arg model)" />
       <arg name="y" value="0.0" />
     </include>
   <node pkg="control" name="posecontrol3" type="posecontrol3"/>
   </group>

   <group ns="uav2">
     <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
       <arg name="name" value="uav2" />
       <arg name="tf_prefix" value="uav2" />
       <arg name="model" value="$(arg model)" />
       <arg name="y" value="1.0" />
     </include>
   <node pkg="control" name="posecontrol3" type="posecontrol3"/>
   </group>

   <group ns="uav3">
     <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
       <arg name="name" value="uav3" />
       <arg name="tf_prefix" value="uav3" />
       <arg name="model" value="$(arg model)" />
       <arg name="y" value="2.0" />
     </include>
   <node pkg="control" name="posecontrol3" type="posecontrol3"/>
   </group>

   <group ns="PC">
     <node pkg="formationbag" name="Formation_leader" type="Formation_leader"/>
     <node pkg="formationbag" name="horizon_ctrl" type="horizon_ctrl" output="screen"/>
     <node pkg="formationbag" name="vertical_ctrl" type="vertical_ctrl" output="screen"/>
     <node pkg="formationbag" name="circle_ctrl" type="circle_ctrl" output="screen"/>     
   </group>


</launch>

